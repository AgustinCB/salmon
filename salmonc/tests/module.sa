mod test {
    var variable = 1;
    variable = variable + 1;
    var variable1 = "eeeh?";

    fun test() {
        print "hola";
    }

    class TestClass {
        testmethod() {
            print "hola method";
            print variable;
        }
    }

    trait TestTrait {
        echo(a);
    }

    fun test1() {
        print variable;
    }
}

print test::variable1;
print test::variable;

test::test();

trait MyTrait {
    mytraitmethod();
}

class Impl < test::TestClass {}

trait MyTrait for test::TestClass {
    mytraitmethod() {
        print "my trait method";
    }
}

var c = test::TestClass();
c.testmethod();
c.mytraitmethod();

trait test::TestTrait for Impl {
  echo(a) {
      print a;
  }
}

var i = Impl();
i.echo("pepe");
i.testmethod();

print test::variable;
test::test1();
