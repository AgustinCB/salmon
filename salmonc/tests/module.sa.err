Constants: [SimpleValue(Function { ip: 432, arity: 0, uplifts: None }), SimpleValue(Integer(1)), SimpleValue(Integer(4)), SimpleValue(Integer(3)), SimpleValue(String(0)), SimpleValue(String(1)), SimpleValue(Nil), SimpleValue(Function { ip: 446, arity: 1, uplifts: None }), SimpleValue(String(5)), SimpleValue(String(16)), SimpleValue(String(26)), SimpleValue(Object { address: 275, tags: 387 }), SimpleValue(String(36)), SimpleValue(String(55)), SimpleValue(Function { ip: 472, arity: 0, uplifts: None }), SimpleValue(Object { address: 406, tags: 470 }), SimpleValue(String(61)), SimpleValue(String(68)), SimpleValue(Function { ip: 486, arity: 1, uplifts: None }), SimpleValue(String(73)), SimpleValue(Function { ip: 500, arity: 2, uplifts: None }), SimpleValue(String(88)), SimpleValue(String(92)), SimpleValue(Integer(0)), SimpleValue(String(98)), SimpleValue(String(111)), SimpleValue(String(116)), SimpleValue(String(120)), SimpleValue(String(130)), SimpleValue(String(140)), SimpleValue(String(158)), SimpleValue(String(176)), SimpleValue(String(197)), SimpleValue(String(218)), SimpleValue(String(228)), SimpleValue(String(247)), SimpleValue(String(251)), SimpleValue(String(263)), SimpleValue(String(477))]
Instructions: [Instruction { instruction_type: Constant(0), location: 1 }, Instruction { instruction_type: SetGlobal(4), location: 1 }, Instruction { instruction_type: Pop, location: 1 }, Instruction { instruction_type: Constant(7), location: 4 }, Instruction { instruction_type: SetGlobal(5), location: 4 }, Instruction { instruction_type: Pop, location: 4 }, Instruction { instruction_type: Constant(12), location: 4 }, Instruction { instruction_type: Constant(13), location: 4 }, Instruction { instruction_type: Constant(11), location: 4 }, Instruction { instruction_type: AddTag, location: 4 }, Instruction { instruction_type: AddTag, location: 4 }, Instruction { instruction_type: SetGlobal(6), location: 4 }, Instruction { instruction_type: Pop, location: 4 }, Instruction { instruction_type: Constant(14), location: 6 }, Instruction { instruction_type: SetGlobal(7), location: 6 }, Instruction { instruction_type: Pop, location: 6 }, Instruction { instruction_type: Constant(16), location: 7 }, Instruction { instruction_type: Constant(13), location: 7 }, Instruction { instruction_type: Constant(15), location: 7 }, Instruction { instruction_type: AddTag, location: 7 }, Instruction { instruction_type: AddTag, location: 7 }, Instruction { instruction_type: Constant(13), location: 7 }, Instruction { instruction_type: GetGlobal(6), location: 7 }, Instruction { instruction_type: RemoveTag, location: 7 }, Instruction { instruction_type: Call, location: 7 }, Instruction { instruction_type: Swap, location: 7 }, Instruction { instruction_type: Constant(17), location: 7 }, Instruction { instruction_type: Swap, location: 7 }, Instruction { instruction_type: ObjectSet, location: 7 }, Instruction { instruction_type: Swap, location: 7 }, Instruction { instruction_type: ObjectMerge, location: 7 }, Instruction { instruction_type: SetGlobal(8), location: 7 }, Instruction { instruction_type: Pop, location: 7 }, Instruction { instruction_type: Constant(18), location: 9 }, Instruction { instruction_type: SetGlobal(9), location: 9 }, Instruction { instruction_type: Pop, location: 9 }, Instruction { instruction_type: Constant(20), location: 11 }, Instruction { instruction_type: SetGlobal(10), location: 11 }, Instruction { instruction_type: Pop, location: 11 }, Instruction { instruction_type: Constant(22), location: 12 }, Instruction { instruction_type: Constant(23), location: 12 }, Instruction { instruction_type: ObjectAlloc, location: 12 }, Instruction { instruction_type: AddTag, location: 12 }, Instruction { instruction_type: SetGlobal(11), location: 12 }, Instruction { instruction_type: Pop, location: 12 }, Instruction { instruction_type: Constant(22), location: 12 }, Instruction { instruction_type: Constant(23), location: 12 }, Instruction { instruction_type: ObjectAlloc, location: 12 }, Instruction { instruction_type: AddTag, location: 12 }, Instruction { instruction_type: SetGlobal(12), location: 12 }, Instruction { instruction_type: Pop, location: 12 }, Instruction { instruction_type: Constant(22), location: 13 }, Instruction { instruction_type: Constant(23), location: 13 }, Instruction { instruction_type: ObjectAlloc, location: 13 }, Instruction { instruction_type: AddTag, location: 13 }, Instruction { instruction_type: SetGlobal(13), location: 13 }, Instruction { instruction_type: Pop, location: 13 }, Instruction { instruction_type: Constant(1), location: 14 }, Instruction { instruction_type: SetGlobal(0), location: 14 }, Instruction { instruction_type: Pop, location: 14 }, Instruction { instruction_type: GetGlobal(0), location: 15 }, Instruction { instruction_type: Constant(1), location: 15 }, Instruction { instruction_type: Plus, location: 15 }, Instruction { instruction_type: SetGlobal(0), location: 15 }, Instruction { instruction_type: Pop, location: 15 }, Instruction { instruction_type: Constant(25), location: 16 }, Instruction { instruction_type: SetGlobal(1), location: 16 }, Instruction { instruction_type: Pop, location: 16 }, Instruction { instruction_type: Constant(6), location: 17 }, Instruction { instruction_type: Pop, location: 17 }, Instruction { instruction_type: Constant(6), location: 17 }, Instruction { instruction_type: Duplicate, location: 17 }, Instruction { instruction_type: JmpIfFalse(4), location: 17 }, Instruction { instruction_type: Constant(10), location: 17 }, Instruction { instruction_type: Constant(11), location: 17 }, Instruction { instruction_type: ObjectSet, location: 17 }, Instruction { instruction_type: Pop, location: 17 }, Instruction { instruction_type: Pop, location: 17 }, Instruction { instruction_type: Constant(6), location: 17 }, Instruction { instruction_type: Pop, location: 17 }, Instruction { instruction_type: Constant(6), location: 17 }, Instruction { instruction_type: Pop, location: 17 }, Instruction { instruction_type: Constant(4), location: 18 }, Instruction { instruction_type: GetGlobal(1), location: 18 }, Instruction { instruction_type: ToStr, location: 18 }, Instruction { instruction_type: StringConcat, location: 18 }, Instruction { instruction_type: Push, location: 18 }, Instruction { instruction_type: Strlen, location: 18 }, Instruction { instruction_type: Swap, location: 18 }, Instruction { instruction_type: Constant(1), location: 18 }, Instruction { instruction_type: Constant(3), location: 18 }, Instruction { instruction_type: Constant(2), location: 18 }, Instruction { instruction_type: Syscall, location: 18 }, Instruction { instruction_type: Pop, location: 18 }, Instruction { instruction_type: Constant(4), location: 19 }, Instruction { instruction_type: GetGlobal(0), location: 19 }, Instruction { instruction_type: ToStr, location: 19 }, Instruction { instruction_type: StringConcat, location: 19 }, Instruction { instruction_type: Push, location: 19 }, Instruction { instruction_type: Strlen, location: 19 }, Instruction { instruction_type: Swap, location: 19 }, Instruction { instruction_type: Constant(1), location: 19 }, Instruction { instruction_type: Constant(3), location: 19 }, Instruction { instruction_type: Constant(2), location: 19 }, Instruction { instruction_type: Syscall, location: 19 }, Instruction { instruction_type: Pop, location: 19 }, Instruction { instruction_type: GetGlobal(4), location: 20 }, Instruction { instruction_type: Duplicate, location: 20 }, Instruction { instruction_type: CheckType(7), location: 20 }, Instruction { instruction_type: JmpIfFalse(19), location: 20 }, Instruction { instruction_type: Constant(26), location: 20 }, Instruction { instruction_type: Swap, location: 20 }, Instruction { instruction_type: Constant(13), location: 20 }, Instruction { instruction_type: Swap, location: 20 }, Instruction { instruction_type: RemoveTag, location: 20 }, Instruction { instruction_type: Call, location: 20 }, Instruction { instruction_type: ObjectHas, location: 20 }, Instruction { instruction_type: JmpIfFalse(12), location: 20 }, Instruction { instruction_type: Duplicate, location: 20 }, Instruction { instruction_type: Constant(26), location: 20 }, Instruction { instruction_type: Swap, location: 20 }, Instruction { instruction_type: ObjectGet, location: 20 }, Instruction { instruction_type: Swap, location: 20 }, Instruction { instruction_type: SetGlobal(14), location: 20 }, Instruction { instruction_type: Pop, location: 20 }, Instruction { instruction_type: Call, location: 20 }, Instruction { instruction_type: Pop, location: 20 }, Instruction { instruction_type: GetGlobal(14), location: 20 }, Instruction { instruction_type: Jmp(1), location: 20 }, Instruction { instruction_type: Call, location: 20 }, Instruction { instruction_type: Pop, location: 20 }, Instruction { instruction_type: GetGlobal(9), location: 20 }, Instruction { instruction_type: Constant(24), location: 20 }, Instruction { instruction_type: GetGlobal(6), location: 20 }, Instruction { instruction_type: ObjectSet, location: 20 }, Instruction { instruction_type: Swap, location: 20 }, Instruction { instruction_type: Pop, location: 20 }, Instruction { instruction_type: SetGlobal(6), location: 20 }, Instruction { instruction_type: Pop, location: 20 }, Instruction { instruction_type: Constant(28), location: 20 }, Instruction { instruction_type: GetGlobal(6), location: 20 }, Instruction { instruction_type: AddTag, location: 20 }, Instruction { instruction_type: SetGlobal(6), location: 20 }, Instruction { instruction_type: Pop, location: 20 }, Instruction { instruction_type: GetGlobal(6), location: 21 }, Instruction { instruction_type: Duplicate, location: 21 }, Instruction { instruction_type: CheckType(7), location: 21 }, Instruction { instruction_type: JmpIfFalse(19), location: 21 }, Instruction { instruction_type: Constant(26), location: 21 }, Instruction { instruction_type: Swap, location: 21 }, Instruction { instruction_type: Constant(13), location: 21 }, Instruction { instruction_type: Swap, location: 21 }, Instruction { instruction_type: RemoveTag, location: 21 }, Instruction { instruction_type: Call, location: 21 }, Instruction { instruction_type: ObjectHas, location: 21 }, Instruction { instruction_type: JmpIfFalse(12), location: 21 }, Instruction { instruction_type: Duplicate, location: 21 }, Instruction { instruction_type: Constant(26), location: 21 }, Instruction { instruction_type: Swap, location: 21 }, Instruction { instruction_type: ObjectGet, location: 21 }, Instruction { instruction_type: Swap, location: 21 }, Instruction { instruction_type: SetGlobal(14), location: 21 }, Instruction { instruction_type: Pop, location: 21 }, Instruction { instruction_type: Call, location: 21 }, Instruction { instruction_type: Pop, location: 21 }, Instruction { instruction_type: GetGlobal(14), location: 21 }, Instruction { instruction_type: Jmp(1), location: 21 }, Instruction { instruction_type: Call, location: 21 }, Instruction { instruction_type: SetGlobal(2), location: 21 }, Instruction { instruction_type: Pop, location: 21 }, Instruction { instruction_type: GetGlobal(2), location: 22 }, Instruction { instruction_type: Duplicate, location: 22 }, Instruction { instruction_type: Constant(13), location: 22 }, Instruction { instruction_type: Swap, location: 22 }, Instruction { instruction_type: CheckTag, location: 22 }, Instruction { instruction_type: JmpIfFalse(4), location: 22 }, Instruction { instruction_type: Constant(29), location: 22 }, Instruction { instruction_type: Swap, location: 22 }, Instruction { instruction_type: ObjectGet, location: 22 }, Instruction { instruction_type: Jmp(12), location: 22 }, Instruction { instruction_type: Constant(30), location: 22 }, Instruction { instruction_type: Swap, location: 22 }, Instruction { instruction_type: ObjectHas, location: 22 }, Instruction { instruction_type: JmpIfFalse(5), location: 22 }, Instruction { instruction_type: Constant(30), location: 22 }, Instruction { instruction_type: Swap, location: 22 }, Instruction { instruction_type: ObjectGet, location: 22 }, Instruction { instruction_type: Call, location: 22 }, Instruction { instruction_type: Jmp(3), location: 22 }, Instruction { instruction_type: Constant(10), location: 22 }, Instruction { instruction_type: Swap, location: 22 }, Instruction { instruction_type: ObjectGet, location: 22 }, Instruction { instruction_type: Duplicate, location: 22 }, Instruction { instruction_type: CheckType(7), location: 22 }, Instruction { instruction_type: JmpIfFalse(19), location: 22 }, Instruction { instruction_type: Constant(26), location: 22 }, Instruction { instruction_type: Swap, location: 22 }, Instruction { instruction_type: Constant(13), location: 22 }, Instruction { instruction_type: Swap, location: 22 }, Instruction { instruction_type: RemoveTag, location: 22 }, Instruction { instruction_type: Call, location: 22 }, Instruction { instruction_type: ObjectHas, location: 22 }, Instruction { instruction_type: JmpIfFalse(12), location: 22 }, Instruction { instruction_type: Duplicate, location: 22 }, Instruction { instruction_type: Constant(26), location: 22 }, Instruction { instruction_type: Swap, location: 22 }, Instruction { instruction_type: ObjectGet, location: 22 }, Instruction { instruction_type: Swap, location: 22 }, Instruction { instruction_type: SetGlobal(14), location: 22 }, Instruction { instruction_type: Pop, location: 22 }, Instruction { instruction_type: Call, location: 22 }, Instruction { instruction_type: Pop, location: 22 }, Instruction { instruction_type: GetGlobal(14), location: 22 }, Instruction { instruction_type: Jmp(1), location: 22 }, Instruction { instruction_type: Call, location: 22 }, Instruction { instruction_type: Pop, location: 22 }, Instruction { instruction_type: GetGlobal(2), location: 23 }, Instruction { instruction_type: Duplicate, location: 23 }, Instruction { instruction_type: Constant(13), location: 23 }, Instruction { instruction_type: Swap, location: 23 }, Instruction { instruction_type: CheckTag, location: 23 }, Instruction { instruction_type: JmpIfFalse(4), location: 23 }, Instruction { instruction_type: Constant(31), location: 23 }, Instruction { instruction_type: Swap, location: 23 }, Instruction { instruction_type: ObjectGet, location: 23 }, Instruction { instruction_type: Jmp(12), location: 23 }, Instruction { instruction_type: Constant(32), location: 23 }, Instruction { instruction_type: Swap, location: 23 }, Instruction { instruction_type: ObjectHas, location: 23 }, Instruction { instruction_type: JmpIfFalse(5), location: 23 }, Instruction { instruction_type: Constant(32), location: 23 }, Instruction { instruction_type: Swap, location: 23 }, Instruction { instruction_type: ObjectGet, location: 23 }, Instruction { instruction_type: Call, location: 23 }, Instruction { instruction_type: Jmp(3), location: 23 }, Instruction { instruction_type: Constant(24), location: 23 }, Instruction { instruction_type: Swap, location: 23 }, Instruction { instruction_type: ObjectGet, location: 23 }, Instruction { instruction_type: Duplicate, location: 23 }, Instruction { instruction_type: CheckType(7), location: 23 }, Instruction { instruction_type: JmpIfFalse(19), location: 23 }, Instruction { instruction_type: Constant(26), location: 23 }, Instruction { instruction_type: Swap, location: 23 }, Instruction { instruction_type: Constant(13), location: 23 }, Instruction { instruction_type: Swap, location: 23 }, Instruction { instruction_type: RemoveTag, location: 23 }, Instruction { instruction_type: Call, location: 23 }, Instruction { instruction_type: ObjectHas, location: 23 }, Instruction { instruction_type: JmpIfFalse(12), location: 23 }, Instruction { instruction_type: Duplicate, location: 23 }, Instruction { instruction_type: Constant(26), location: 23 }, Instruction { instruction_type: Swap, location: 23 }, Instruction { instruction_type: ObjectGet, location: 23 }, Instruction { instruction_type: Swap, location: 23 }, Instruction { instruction_type: SetGlobal(14), location: 23 }, Instruction { instruction_type: Pop, location: 23 }, Instruction { instruction_type: Call, location: 23 }, Instruction { instruction_type: Pop, location: 23 }, Instruction { instruction_type: GetGlobal(14), location: 23 }, Instruction { instruction_type: Jmp(1), location: 23 }, Instruction { instruction_type: Call, location: 23 }, Instruction { instruction_type: Pop, location: 23 }, Instruction { instruction_type: GetGlobal(10), location: 23 }, Instruction { instruction_type: Constant(21), location: 23 }, Instruction { instruction_type: GetGlobal(8), location: 23 }, Instruction { instruction_type: ObjectSet, location: 23 }, Instruction { instruction_type: Swap, location: 23 }, Instruction { instruction_type: Pop, location: 23 }, Instruction { instruction_type: SetGlobal(8), location: 23 }, Instruction { instruction_type: Pop, location: 23 }, Instruction { instruction_type: Constant(34), location: 23 }, Instruction { instruction_type: GetGlobal(8), location: 23 }, Instruction { instruction_type: AddTag, location: 23 }, Instruction { instruction_type: SetGlobal(8), location: 23 }, Instruction { instruction_type: Pop, location: 23 }, Instruction { instruction_type: GetGlobal(8), location: 24 }, Instruction { instruction_type: Duplicate, location: 24 }, Instruction { instruction_type: CheckType(7), location: 24 }, Instruction { instruction_type: JmpIfFalse(19), location: 24 }, Instruction { instruction_type: Constant(26), location: 24 }, Instruction { instruction_type: Swap, location: 24 }, Instruction { instruction_type: Constant(13), location: 24 }, Instruction { instruction_type: Swap, location: 24 }, Instruction { instruction_type: RemoveTag, location: 24 }, Instruction { instruction_type: Call, location: 24 }, Instruction { instruction_type: ObjectHas, location: 24 }, Instruction { instruction_type: JmpIfFalse(12), location: 24 }, Instruction { instruction_type: Duplicate, location: 24 }, Instruction { instruction_type: Constant(26), location: 24 }, Instruction { instruction_type: Swap, location: 24 }, Instruction { instruction_type: ObjectGet, location: 24 }, Instruction { instruction_type: Swap, location: 24 }, Instruction { instruction_type: SetGlobal(14), location: 24 }, Instruction { instruction_type: Pop, location: 24 }, Instruction { instruction_type: Call, location: 24 }, Instruction { instruction_type: Pop, location: 24 }, Instruction { instruction_type: GetGlobal(14), location: 24 }, Instruction { instruction_type: Jmp(1), location: 24 }, Instruction { instruction_type: Call, location: 24 }, Instruction { instruction_type: SetGlobal(3), location: 24 }, Instruction { instruction_type: Pop, location: 24 }, Instruction { instruction_type: Constant(35), location: 25 }, Instruction { instruction_type: GetGlobal(3), location: 25 }, Instruction { instruction_type: Duplicate, location: 25 }, Instruction { instruction_type: Constant(13), location: 25 }, Instruction { instruction_type: Swap, location: 25 }, Instruction { instruction_type: CheckTag, location: 25 }, Instruction { instruction_type: JmpIfFalse(4), location: 25 }, Instruction { instruction_type: Constant(36), location: 25 }, Instruction { instruction_type: Swap, location: 25 }, Instruction { instruction_type: ObjectGet, location: 25 }, Instruction { instruction_type: Jmp(12), location: 25 }, Instruction { instruction_type: Constant(37), location: 25 }, Instruction { instruction_type: Swap, location: 25 }, Instruction { instruction_type: ObjectHas, location: 25 }, Instruction { instruction_type: JmpIfFalse(5), location: 25 }, Instruction { instruction_type: Constant(37), location: 25 }, Instruction { instruction_type: Swap, location: 25 }, Instruction { instruction_type: ObjectGet, location: 25 }, Instruction { instruction_type: Call, location: 25 }, Instruction { instruction_type: Jmp(3), location: 25 }, Instruction { instruction_type: Constant(21), location: 25 }, Instruction { instruction_type: Swap, location: 25 }, Instruction { instruction_type: ObjectGet, location: 25 }, Instruction { instruction_type: Duplicate, location: 25 }, Instruction { instruction_type: CheckType(7), location: 25 }, Instruction { instruction_type: JmpIfFalse(19), location: 25 }, Instruction { instruction_type: Constant(26), location: 25 }, Instruction { instruction_type: Swap, location: 25 }, Instruction { instruction_type: Constant(13), location: 25 }, Instruction { instruction_type: Swap, location: 25 }, Instruction { instruction_type: RemoveTag, location: 25 }, Instruction { instruction_type: Call, location: 25 }, Instruction { instruction_type: ObjectHas, location: 25 }, Instruction { instruction_type: JmpIfFalse(12), location: 25 }, Instruction { instruction_type: Duplicate, location: 25 }, Instruction { instruction_type: Constant(26), location: 25 }, Instruction { instruction_type: Swap, location: 25 }, Instruction { instruction_type: ObjectGet, location: 25 }, Instruction { instruction_type: Swap, location: 25 }, Instruction { instruction_type: SetGlobal(14), location: 25 }, Instruction { instruction_type: Pop, location: 25 }, Instruction { instruction_type: Call, location: 25 }, Instruction { instruction_type: Pop, location: 25 }, Instruction { instruction_type: GetGlobal(14), location: 25 }, Instruction { instruction_type: Jmp(1), location: 25 }, Instruction { instruction_type: Call, location: 25 }, Instruction { instruction_type: Pop, location: 25 }, Instruction { instruction_type: GetGlobal(3), location: 26 }, Instruction { instruction_type: Duplicate, location: 26 }, Instruction { instruction_type: Constant(13), location: 26 }, Instruction { instruction_type: Swap, location: 26 }, Instruction { instruction_type: CheckTag, location: 26 }, Instruction { instruction_type: JmpIfFalse(4), location: 26 }, Instruction { instruction_type: Constant(29), location: 26 }, Instruction { instruction_type: Swap, location: 26 }, Instruction { instruction_type: ObjectGet, location: 26 }, Instruction { instruction_type: Jmp(12), location: 26 }, Instruction { instruction_type: Constant(30), location: 26 }, Instruction { instruction_type: Swap, location: 26 }, Instruction { instruction_type: ObjectHas, location: 26 }, Instruction { instruction_type: JmpIfFalse(5), location: 26 }, Instruction { instruction_type: Constant(30), location: 26 }, Instruction { instruction_type: Swap, location: 26 }, Instruction { instruction_type: ObjectGet, location: 26 }, Instruction { instruction_type: Call, location: 26 }, Instruction { instruction_type: Jmp(3), location: 26 }, Instruction { instruction_type: Constant(10), location: 26 }, Instruction { instruction_type: Swap, location: 26 }, Instruction { instruction_type: ObjectGet, location: 26 }, Instruction { instruction_type: Duplicate, location: 26 }, Instruction { instruction_type: CheckType(7), location: 26 }, Instruction { instruction_type: JmpIfFalse(19), location: 26 }, Instruction { instruction_type: Constant(26), location: 26 }, Instruction { instruction_type: Swap, location: 26 }, Instruction { instruction_type: Constant(13), location: 26 }, Instruction { instruction_type: Swap, location: 26 }, Instruction { instruction_type: RemoveTag, location: 26 }, Instruction { instruction_type: Call, location: 26 }, Instruction { instruction_type: ObjectHas, location: 26 }, Instruction { instruction_type: JmpIfFalse(12), location: 26 }, Instruction { instruction_type: Duplicate, location: 26 }, Instruction { instruction_type: Constant(26), location: 26 }, Instruction { instruction_type: Swap, location: 26 }, Instruction { instruction_type: ObjectGet, location: 26 }, Instruction { instruction_type: Swap, location: 26 }, Instruction { instruction_type: SetGlobal(14), location: 26 }, Instruction { instruction_type: Pop, location: 26 }, Instruction { instruction_type: Call, location: 26 }, Instruction { instruction_type: Pop, location: 26 }, Instruction { instruction_type: GetGlobal(14), location: 26 }, Instruction { instruction_type: Jmp(1), location: 26 }, Instruction { instruction_type: Call, location: 26 }, Instruction { instruction_type: Pop, location: 26 }, Instruction { instruction_type: Constant(4), location: 27 }, Instruction { instruction_type: GetGlobal(0), location: 27 }, Instruction { instruction_type: ToStr, location: 27 }, Instruction { instruction_type: StringConcat, location: 27 }, Instruction { instruction_type: Push, location: 27 }, Instruction { instruction_type: Strlen, location: 27 }, Instruction { instruction_type: Swap, location: 27 }, Instruction { instruction_type: Constant(1), location: 27 }, Instruction { instruction_type: Constant(3), location: 27 }, Instruction { instruction_type: Constant(2), location: 27 }, Instruction { instruction_type: Syscall, location: 27 }, Instruction { instruction_type: Pop, location: 27 }, Instruction { instruction_type: GetGlobal(7), location: 28 }, Instruction { instruction_type: Duplicate, location: 28 }, Instruction { instruction_type: CheckType(7), location: 28 }, Instruction { instruction_type: JmpIfFalse(19), location: 28 }, Instruction { instruction_type: Constant(26), location: 28 }, Instruction { instruction_type: Swap, location: 28 }, Instruction { instruction_type: Constant(13), location: 28 }, Instruction { instruction_type: Swap, location: 28 }, Instruction { instruction_type: RemoveTag, location: 28 }, Instruction { instruction_type: Call, location: 28 }, Instruction { instruction_type: ObjectHas, location: 28 }, Instruction { instruction_type: JmpIfFalse(12), location: 28 }, Instruction { instruction_type: Duplicate, location: 28 }, Instruction { instruction_type: Constant(26), location: 28 }, Instruction { instruction_type: Swap, location: 28 }, Instruction { instruction_type: ObjectGet, location: 28 }, Instruction { instruction_type: Swap, location: 28 }, Instruction { instruction_type: SetGlobal(14), location: 28 }, Instruction { instruction_type: Pop, location: 28 }, Instruction { instruction_type: Call, location: 28 }, Instruction { instruction_type: Pop, location: 28 }, Instruction { instruction_type: GetGlobal(14), location: 28 }, Instruction { instruction_type: Jmp(1), location: 28 }, Instruction { instruction_type: Call, location: 28 }, Instruction { instruction_type: Pop, location: 28 }, Instruction { instruction_type: Return, location: 28 }, Instruction { instruction_type: Constant(4), location: 1 }, Instruction { instruction_type: Constant(5), location: 1 }, Instruction { instruction_type: ToStr, location: 1 }, Instruction { instruction_type: StringConcat, location: 1 }, Instruction { instruction_type: Push, location: 1 }, Instruction { instruction_type: Strlen, location: 1 }, Instruction { instruction_type: Swap, location: 1 }, Instruction { instruction_type: Constant(1), location: 1 }, Instruction { instruction_type: Constant(3), location: 1 }, Instruction { instruction_type: Constant(2), location: 1 }, Instruction { instruction_type: Syscall, location: 1 }, Instruction { instruction_type: Pop, location: 1 }, Instruction { instruction_type: Constant(6), location: 1 }, Instruction { instruction_type: Return, location: 1 }, Instruction { instruction_type: Constant(4), location: 3 }, Instruction { instruction_type: Constant(8), location: 3 }, Instruction { instruction_type: ToStr, location: 3 }, Instruction { instruction_type: StringConcat, location: 3 }, Instruction { instruction_type: Push, location: 3 }, Instruction { instruction_type: Strlen, location: 3 }, Instruction { instruction_type: Swap, location: 3 }, Instruction { instruction_type: Constant(1), location: 3 }, Instruction { instruction_type: Constant(3), location: 3 }, Instruction { instruction_type: Constant(2), location: 3 }, Instruction { instruction_type: Syscall, location: 3 }, Instruction { instruction_type: Pop, location: 3 }, Instruction { instruction_type: Constant(4), location: 4 }, Instruction { instruction_type: GetGlobal(0), location: 4 }, Instruction { instruction_type: ToStr, location: 4 }, Instruction { instruction_type: StringConcat, location: 4 }, Instruction { instruction_type: Push, location: 4 }, Instruction { instruction_type: Strlen, location: 4 }, Instruction { instruction_type: Swap, location: 4 }, Instruction { instruction_type: Constant(1), location: 4 }, Instruction { instruction_type: Constant(3), location: 4 }, Instruction { instruction_type: Constant(2), location: 4 }, Instruction { instruction_type: Syscall, location: 4 }, Instruction { instruction_type: Pop, location: 4 }, Instruction { instruction_type: Constant(6), location: 4 }, Instruction { instruction_type: Return, location: 4 }, Instruction { instruction_type: Constant(4), location: 6 }, Instruction { instruction_type: GetGlobal(0), location: 6 }, Instruction { instruction_type: ToStr, location: 6 }, Instruction { instruction_type: StringConcat, location: 6 }, Instruction { instruction_type: Push, location: 6 }, Instruction { instruction_type: Strlen, location: 6 }, Instruction { instruction_type: Swap, location: 6 }, Instruction { instruction_type: Constant(1), location: 6 }, Instruction { instruction_type: Constant(3), location: 6 }, Instruction { instruction_type: Constant(2), location: 6 }, Instruction { instruction_type: Syscall, location: 6 }, Instruction { instruction_type: Pop, location: 6 }, Instruction { instruction_type: Constant(6), location: 6 }, Instruction { instruction_type: Return, location: 6 }, Instruction { instruction_type: Constant(4), location: 9 }, Instruction { instruction_type: Constant(19), location: 9 }, Instruction { instruction_type: ToStr, location: 9 }, Instruction { instruction_type: StringConcat, location: 9 }, Instruction { instruction_type: Push, location: 9 }, Instruction { instruction_type: Strlen, location: 9 }, Instruction { instruction_type: Swap, location: 9 }, Instruction { instruction_type: Constant(1), location: 9 }, Instruction { instruction_type: Constant(3), location: 9 }, Instruction { instruction_type: Constant(2), location: 9 }, Instruction { instruction_type: Syscall, location: 9 }, Instruction { instruction_type: Pop, location: 9 }, Instruction { instruction_type: Constant(6), location: 9 }, Instruction { instruction_type: Return, location: 9 }, Instruction { instruction_type: Constant(4), location: 11 }, Instruction { instruction_type: GetLocal(1), location: 11 }, Instruction { instruction_type: ToStr, location: 11 }, Instruction { instruction_type: StringConcat, location: 11 }, Instruction { instruction_type: Push, location: 11 }, Instruction { instruction_type: Strlen, location: 11 }, Instruction { instruction_type: Swap, location: 11 }, Instruction { instruction_type: Constant(1), location: 11 }, Instruction { instruction_type: Constant(3), location: 11 }, Instruction { instruction_type: Constant(2), location: 11 }, Instruction { instruction_type: Syscall, location: 11 }, Instruction { instruction_type: Pop, location: 11 }, Instruction { instruction_type: Constant(6), location: 11 }, Instruction { instruction_type: Return, location: 11 }]
Locations: [Location { address: 477, line: 5 }, Location { address: 477, line: 6 }, Location { address: 477, line: 9 }, Location { address: 477, line: 11 }, Location { address: 477, line: 12 }, Location { address: 477, line: 20 }, Location { address: 477, line: 21 }, Location { address: 477, line: 34 }, Location { address: 477, line: 36 }, Location { address: 477, line: 38 }, Location { address: 477, line: 46 }, Location { address: 477, line: 48 }, Location { address: 477, line: 16 }, Location { address: 477, line: 30 }, Location { address: 477, line: 1 }, Location { address: 477, line: 2 }, Location { address: 477, line: 3 }, Location { address: 477, line: 0 }, Location { address: 477, line: 25 }, Location { address: 477, line: 26 }, Location { address: 477, line: 28 }, Location { address: 477, line: 42 }, Location { address: 477, line: 43 }, Location { address: 477, line: 44 }, Location { address: 477, line: 52 }, Location { address: 477, line: 53 }, Location { address: 477, line: 54 }, Location { address: 477, line: 56 }, Location { address: 477, line: 57 }]
