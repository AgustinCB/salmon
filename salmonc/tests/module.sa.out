eeeh?
2
hola
hola method
2
my trait method
pepe
hola method
2
2
2
